rust/sedona-schema/src/datatypes.rs

```diff
@@ -333,6 +365,1241 @@ fn deserialize_edges(edges: &Value) -> Result<Edges> {
+            column::BAND,
+            RasterSchema::band_type(),
+            false,
+        ));
+
+        // Now create the main builder with pre-built components
+        let main_builder = StructBuilder::new(
+            RasterSchema::fields(),
+            vec![
+                Box::new(metadata_builder),
+                Box::new(crs_builder),
+                Box::new(bbox_builder),
+                Box::new(bands_builder),
+            ],
+        );
+
+        Self { main_builder }
+    }
+
+    /// Start a new raster with metadata, optional CRS, and optional bounding box
+    ///
+    /// This is the unified method for starting a raster with all optional parameters.
+    ///
+    /// # Arguments
+    /// * `metadata` - Raster metadata (dimensions, geotransform parameters)
+    /// * `crs` - Optional coordinate reference system as string
+    /// * `bbox` - Optional bounding box coordinates
+    ///
+    /// # Examples
+    /// // From iterator - copy all fields from existing raster
+    /// builder.start_raster(raster.metadata(), raster.crs(), raster.bounding_box(0).as_ref())?;
+    ///
+    /// // From RasterMetadata struct with all fields
+    /// builder.start_raster(&metadata, Some("EPSG:4326"), metadata.bounding_box.as_ref())?;
+    ///
+    /// // Minimal - just metadata
+    /// builder.start_raster(&metadata, None, None)?;
+    /// ```
+    pub fn start_raster(
+        &mut self,
+        metadata: &dyn MetadataRef,
+        crs: Option<&str>,
+        bbox: Option<&BoundingBox>,
+    ) -> Result<(), ArrowError> {
+        self.append_metadata_from_ref(metadata)?;
+        self.set_crs(crs)?;
+        self.append_bounding_box(bbox)?;
+        Ok(())
+    }
+
+    /// Get direct access to the BinaryBuilder for writing the current band's data
+    pub fn band_data_writer(&mut self) -> &mut BinaryBuilder {
+        let bands_builder = self
+            .main_builder
+            .field_builder::<ListBuilder<StructBuilder>>(raster_indices::BANDS)
+            .unwrap();
+        let band_builder = bands_builder.values();
+        // Ensure we have at least one field (band metadata and data)
+        // Field 0 = metadata (StructBuilder), Field 1 = data (BinaryBuilder)
+        band_builder.field_builder::<BinaryBuilder>(1).unwrap()
+    }
+
+    /// Create a MutableBuffer that can be written to directly
+    pub fn create_band_buffer(
+        &mut self,
+        capacity: usize,
+    ) -> (MutableBuffer, impl FnOnce(MutableBuffer) + '_) {
+        let mut buffer = MutableBuffer::with_capacity(capacity);
+
+        // Pre-allocate the buffer to the exact size
+        buffer.resize(capacity, 0);
+
+        let commit = move |buffer: MutableBuffer| {
+            // Convert MutableBuffer to &[u8] and append to BinaryBuilder
+            let data = buffer.as_slice();
+            self.band_data_writer().append_value(data);
+        };
+
+        (buffer, commit)
+    }
```

Comment:
We already have another method `commit_band_buffer`, so we don't have to return a committer with the buffer.